charts/opentelemetry-operator/values.schema.json

```diff
@@ -0,0 +1,891 @@
+{
+    "$schema": "http://json-schema.org/schema#",
+    "type": "object",
+    "default": {},
+    "title": "Values",
+    "required": [
+        "replicaCount",
+        "nameOverride",
+        "imagePullSecrets",
+        "manager",
+        "kubeRBACProxy",
+        "admissionWebhooks",
+        "role",
+        "clusterRole",
+        "affinity",
+        "tolerations",
+        "nodeSelector",
+        "hostNetwork"
+    ],
+    "properties": {
```

Comment:
I think we should use  `"additionalProperties": false` at the top level, and then within objects when appropriate.  This is the field that helps use with deprecations or breaking changes in the future.  

With the addition of `additionalProperties` flag at the top level, we should also include 

```yaml
    "enabled": {
      "description": "Usually used when using Opentelemetry-collector as a subchart.",
      "type": "boolean"
    },
    "global": {
      "type": "object"
    },
```
in the properties.  These are fields that we've added in the collector chart to help with a helm defect and subcharts.

Reply:
@TylerHelmuth added